core/getModelFromHomology.m

```diff
@@ -455,7 +455,8 @@
                 for l=2:numel(b)
                     repString=[repString ') or (' fullGeneList{b(l)}];
                 end
-                models{useOrderIndexes(i)}.grRules{j}=strrep(models{useOrderIndexes(i)}.grRules{j},geneName{1},repString);
+                %Use regexprep instead of strrep to prevent partial matches
+                models{useOrderIndexes(i)}.grRules{j}=regexprep(models{useOrderIndexes(i)}.grRules{j},[geneName{1} '($|\s|\))'],[repString '$1']);

```

Comment:
This fix will prevent partial matches for genes with the same starting names, e.g. YAL001W and YAL001W-A, as you claimed. However, it will not prevent partial matches for genes with the same *ending* names, e.g., AGENE and GENE. I would therefore suggest the following line of code:

`models{useOrderIndexes(i)}.grRules{j}=regexprep(models{useOrderIndexes(i)}.grRules{j},['(^|\s|\()' geneName{1} '($|\s|\))'],['$1' repString '$2]);`

Reply:
Thanks @JonathanRob, you're completely right.
When suggesting a code change when reviewing a PR, see [here](https://help.github.com/en/articles/reviewing-proposed-changes-in-a-pull-request) point 6, this makes is very convenient to apply the suggested changes.

Reply:
Ah, very nice, I'll use this next time. This code-commenting feature is still a bit new to me.